src/main/java/org/jalasoft/moi/controller/services/FileService.java

```diff
@@ -68,8 +75,9 @@ public FileCode getFileById(Long id) {
     public FileCode addNewFile(String name, String code, Long projectID) throws IOException {
         FileCode newFilecode = new FileCode();
         newFilecode.setName(name);
-        newFilecode.setCode(saveFile(name,code, projectID));
-        newFilecode.setProject(projectRepository.findById(projectID).get());
+        newFilecode.setProject(projectRepository.findById(projectID).orElse(null));
+        saveFileB64(name, code, projectID);
+        LOGGER.info("Project id retrieved from URL: {}", projectID);
         return fileRepository.save(newFilecode);
```

Comment:
I think the names of the `saveFileB64` and `save` methods are a bit confusing since it seems that they are both doing the same job.
You can rename them or divide the work, one could like `convertB64toFile` and the other like `saveFile`.

Reply:
I will continue using saveFileB64( ) because that method use saveFile, but I will change saveFile in to saveFileToProjectFolder( )